src/util/favorites.js

```diff
@@ -120,22 +122,32 @@ const models2objects = config => {
         config.rows = config.rows.map(cleanDimension);
     }
 
-    if (config.params) {
-        const { datasetId, params, filter, periodName } = config;
+    if (layer === 'earthEngine') {
+        const { datasetId: id, band, params, aggregationType, filter } = config;
 
-        // EE layer config
-        config.config = JSON.stringify({
-            id: datasetId,
+        const eeConfig = {
+            id,
             params,
-            image: filter ? filter[0].arguments[1] : null,
+            band,
+            aggregationType,
             filter,
-            periodName,
-        });
+        };
+
+        // Removes undefined keys before stringify
+        config.config = JSON.stringify(
+            Object.keys(eeConfig).forEach(
```

Comment:
Does this work?
Isn't `forEach` returning `undefined`?

Reply:
forEach is returning the object keys which are not undefined.

Reply:
🤔  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#return_value

Reply:
Fixed in #1477 as this branch was already merged.